DuplicateFinderEngine/ScanEngine.cs

```diff
@@ -439,12 +452,12 @@ private static class ExtensionMethods {
 			}
 			[MethodImpl(MethodImplOptions.AggressiveInlining)]
 			public static float PercentageDifference2(byte[] img1, byte[] img2) {
-				if (img1.AsSpan().SequenceEqual(img2.AsSpan())) return 1f;
-				float diff = 0;
+				if (img1.Length != img2.Length) throw new Exception("Images must be of same length");
```

Comment:
This is redundant, we know the bytes have equal length

Reply:
future proofing ;) or atleast a better exception message than IndexOutOfRange, or a silent failure if the img1 < img2.
anyway, the length field of an array is right [at the start](https://stackoverflow.com/questions/487202/memory-layout-of-a-net-array) of the array object, so computationally it should take as much cycles as if the array was 4 bytes longer.

but feel free to remove it ;)

Reply:
Maybe doing an Debug.assert? I believe this way its getting removed in release build but throws in debug mode, which we are using when doing code changes.